src/main/java/sg/dex/starfish/impl/file/FileAsset.java

```diff
@@ -23,96 +21,135 @@
+     * @param isHashOfContentRequired if true then hash of content is calculated and included in metadata.
+     *                                Hash of content will be calculated using Keccak256 hash function
+     *                                if false then content hash is not included in metadata
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, String metaData, boolean isHashOfContentRequired) {
+        return new FileAsset(buildMetadata(f, metaData, isHashOfContentRequired), f);
+    }
+
+    /**
+     * This method is to crete File Asset with specific given path.
+     *
+     * @param f path of the file
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f) {
+        return new FileAsset(buildMetadata(f, null, false), f);
+    }
+
+    /**
+     * This method is to crete Resource Asset with specific given path, metadata
+     *
+     * @param f        path of the file
+     * @param metaData metadata associated with the asset.This metadata will be be added in addition to default
+     *                 metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *                 default value will be overridden.
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, String metaData) {
+        return new FileAsset(buildMetadata(f, metaData, false), f);
+    }
+
+    /**
+     * Build default metadata for a file asset
+     *
+     * @param f    The file to use for this file asset
+     * @param meta metadata associated with the asset.This metadata will be be added in addition to default
+     *             metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *             default value will be overridden.
+     * @return The default metadata as a String
+     */
+    protected static String buildMetadata(File f, String meta, boolean isHashOfContentRequired) {
+
+        Map<String, Object> ob = Utils.createDefaultMetadata();
+        ob.put(TYPE, DATA_SET);
+        ob.put(CONTENT_TYPE, OCTET_STREAM);
+
+        if (meta != null) {
+            Map<String, Object> metaMap = JSON.toMap(meta);
+            for (Map.Entry<String, Object> me : metaMap.entrySet()) {
+                ob.put(me.getKey(), me.getValue());
+            }
+        }
+        if (isHashOfContentRequired) {
+            ob.put(CONTENT_HASH, getHashContent(f));
+        }
+
+        return JSON.toString(ob);
+
+    }
+    /**
+     * This method is used to calculate the hash of the content by using keccak256 hashing algorithm.
+     *
+     * @param f path of the file
+     * @return the content of hash as string
+     */
+
+    private static String getHashContent(File f) {
+        String content = null;
+        try {
+            content = Utils.stringFromStream(new FileInputStream(f));
```

Comment:
This is wrong. file content won't be a String in general. Don't try and convert it to one